bin/emqx

```diff
@@ -292,6 +299,33 @@ bootstrapd() {
     fi
 }
 
+# check if a PID is down
+is_down() {
+    PID="$1"
+    if kill -s 0 "$PID" 2>/dev/null; then
+        return 1
+    fi
+    return 0
+}
+
+wait_for() {
+    local WAIT_TIME
+    local CMD
+    WAIT_TIME="$1"
+    shift
+    CMD="$*"
+    while true; do
```

Comment:
can bash `timeout` help here?

Reply:
it's a command which may not exist in e.g. macos?
also here the probes don't really take a long time, but it's sleep.
and `timeout` can not be used on bash functions, only commands